From 463d5e1d2b1ebd90f6b551b71993bc29b2b72842 Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Thu, 5 May 2011 14:32:09 -0700
Subject: build: Add support for droidboot environment

We conditionally pull in the Droidboot ramdisk makefile if necessary.

Actual products will likely want to conditionally enable
TARGET_USE_DROIDBOOT depending on build type, with it not
enabled for 'user' builds.

Change-Id: I516bf7d7049df59ddea245f613a4fc33946fa0af
---
 core/Makefile      |    4 ++++
 core/cleanbuild.mk |    1 +
 2 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index b651eec..f91000f 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -359,6 +359,10 @@ event-log-tags: $(event_log_tags_file)
 ALL_DEFAULT_INSTALLED_MODULES += $(event_log_tags_file)
 
 
+ifeq ($(strip $(TARGET_USE_DROIDBOOT)),true)
+include bootable/droidboot/ramdisk.mk
+endif
+
 # #################################################################
 # Targets for boot/OS images
 # #################################################################
diff --git a/core/cleanbuild.mk b/core/cleanbuild.mk
index 80a46c9..0a0a2bb 100644
--- a/core/cleanbuild.mk
+++ b/core/cleanbuild.mk
@@ -196,6 +196,7 @@ installclean_files := \
 	$(PRODUCT_OUT)/obj/kernelmods \
 	$(PRODUCT_OUT)/System.map \
 	$(PRODUCT_OUT)/kernel \
+	$(PRODUCT_OUT)/droidboot \
 	$(PRODUCT_OUT)/obj/FAKE \
 	$(PRODUCT_OUT)/obj/EXECUTABLES/adbd_intermediates \
 	$(PRODUCT_OUT)/obj/EXECUTABLES/init_intermediates \
-- 
1.7.4.1

